### PR TITLE
Fix AutoComplete onClean/onSelect event

### PR DESCRIPTION
**Pull request by allyssonsantos**
This PR fixes the issue that "onSelect" event was being called when the user clicks on the "Clear (X)" button.

|Before|After|
|------|-----|
|autocomplete-before|autocomplete-after|

### CODE REVIEW

**Review by coelhucas — APPROVED** [review]
Well documented :)

File: packages/yoga/src/AutoComplete/web/AutoComplete.jsx [hunk 1]
```diff
@@ -159,14 +159,19 @@ const AutoComplete = ({
     <Downshift
       selectedItem={userValue}
       onStateChange={changes => {
-        if (Object.prototype.hasOwnProperty.call(changes, 'selectedItem')) {
-          setUserValue(changes.selectedItem);
-          onSelect(changes.selectedItem);
+        const { selectedItem, inputValue } = changes;
+        if (
+          Object.prototype.hasOwnProperty.call(changes, 'selectedItem') &&
+          selectedItem
```
Comment (tcK1) [inline packages/yoga/src/AutoComplete/web/AutoComplete.jsx:165]: Is there a difference between `Object.prototype.hasOwnProperty.call(changes, 'selectedItem')` and `Boolean(changes.selectedItem)` in this case? Same applies to the `inputValue` validation.

What would happen if we just used then directly?

```js
if (selectedItem) {
  /* ... */
} else if (inputValue) {
  /* ... */
}
```

File: packages/yoga/src/AutoComplete/web/AutoComplete.test.jsx [hunk 2]
```diff
@@ -116,6 +124,33 @@ describe('<AutoComplete />', () => {
       fireEvent.click(getByRole('button'));
 
       expect(onCleanMock).toHaveBeenCalledWith('');
+      expect(onSelectMock).not.toHaveBeenCalled();
+      expect(onChangeMock).toHaveBeenCalled();
```
Comment (tcK1) [inline packages/yoga/src/AutoComplete/web/AutoComplete.test.jsx:128]: Should we use `toHaveBeenCalledWith` on these specs to assert if we are passing the correct values to the callback functions?

Comment (allyssonsantos) [reply packages/yoga/src/AutoComplete/web/AutoComplete.test.jsx:128]: yes! well noticed, thanks!

File: packages/yoga/src/AutoComplete/web/AutoComplete.jsx (hunk 1 above)
Comment (allyssonsantos) [reply packages/yoga/src/AutoComplete/web/AutoComplete.jsx:165]: When a user types or selects some suggestion on the field, `changes` object will have "selectedItem" and/or "inputValue" keys inside it.

Talking about "selectedItem" it doesn't make any difference using hasOwnProperty or not.

When we talk about `inputValue` we must call `onChange` even if the "inputValue" is an empty string (like if the user clicks on the "clean" button or deletes the whole string). So, we need to make sure that "inputValue" key exists (even with a falsy value).

**Review by tcK1 — APPROVED** [review]
🚀

File: packages/yoga/src/AutoComplete/web/AutoComplete.jsx (hunk 1 above)
Comment (tcK1) [reply packages/yoga/src/AutoComplete/web/AutoComplete.jsx]: That's perfect.

**Pull request merged**